### PR TITLE
Add feature-flagged Underground Trending

### PR DESCRIPTION
**Pull request by piazzatron**
### Description
- This feature is flagged to be enabled only on staging.

### Dragons

Had to change remote config hook slightly. Shouldn't have any weird effects.

### How Has This Been Tested?

Locally, one window running staging one running prod.

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/piazz-underground-trending

File: src/containers/explore-page/collections.ts [hunk 1]
```diff
@@ -48,13 +51,25 @@ export const TOP_ALBUMS: ExploreCollection = {
...
+  incentivized: true
+}
+
+export const TRENDING_UNDERGROUND: ExploreCollection = {
+  variant: ExploreCollectionsVariant.DIRECT_LINK,
+  title: 'Underground Trending',
+  subtitle: 'Some of the best up-and-coming music on Audius all in one place',
+  gradient: 'linear-gradient(315deg, #BA27FF 0%, #EF8CD9 100%)',
+  shadow: 'rgba(242, 87, 255, 0.35)',
+  icon: IconCassette,
```
Comment (raymondjacobson) [inline src/containers/explore-page/collections.ts:70]: what a good icon

File: src/containers/explore-page/components/desktop/ExplorePage.tsx [hunk 2]
```diff
@@ -122,9 +134,13 @@ const ExplorePage = ({
       <Section
         title={messages.justForYou}
         subtitle={messages.justForYouSubtitle}
-        layout={Layout.TWO_COLUMN_DYNAMIC_WITH_LEADING_ELEMENT}
+        layout={
+          isUnderground
+            ? Layout.TWO_COLUMN_DYNAMIC_WITH_DOUBLE_LEADING_ELEMENT
```
Comment (raymondjacobson) [inline src/containers/explore-page/components/desktop/ExplorePage.tsx:139]: overall the remote var was super clean, except this special casing here. a bit scary. wonder if we could be more pragmatic about the arrangement based on the count length or something

File: src/containers/trending-underground/TrendingUndergroundPage.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+const useLineupReset = () => {
+  const dispatch = useDispatch()
+  useEffect(() => {
+    return () => {
+      dispatch(trendingUndergroundLineupActions.reset())
+    }
+  }, [dispatch])
+}
+
+export const useIsTrendingUndergroundEnabled = () => {
```
Comment (raymondjacobson) [inline src/containers/trending-underground/TrendingUndergroundPage.tsx:100]: I wonder if we should define these hooks nearer to useFlag. might be easier to find all the uses of a particular flag when going through remove later on

File: src/containers/explore-page/collections.ts [hunk 4]
```diff
@@ -109,8 +124,7 @@ export const INTENSE_PLAYLISTS: ExploreMoodCollection = {
 
 export const EXPLORE_COLLECTIONS_MAP = {
   [ExploreCollectionsVariant.LET_THEM_DJ]: LET_THEM_DJ,
-  [ExploreCollectionsVariant.TOP_ALBUMS]: TOP_ALBUMS,
-  [ExploreCollectionsVariant.TRENDING_PLAYLISTS]: TRENDING_PLAYLISTS
```
Comment (raymondjacobson) [inline src/containers/explore-page/collections.ts:113]: was this just a typo? / not needed

**Review by raymondjacobson — APPROVED** [review]
📈

File: src/containers/explore-page/components/desktop/ExplorePage.tsx (hunk 2 above)
Comment (piazzatron) [reply src/containers/explore-page/components/desktop/ExplorePage.tsx:139]: I'll sync with you about this on slack - I went through a number of different iterations of this lol

File: src/containers/trending-underground/TrendingUndergroundPage.tsx (hunk 3 above)
Comment (piazzatron) [reply src/containers/trending-underground/TrendingUndergroundPage.tsx:100]: Hmm interesting - my gameplan for removing this was to just delete the hook and then see all the places where it fails to compile. I don't super love the idea of the remote config folder growing as we add more features, beyond the need to have defaults etc. Think I'll stick with colocating for now

File: src/containers/explore-page/collections.ts (hunk 4 above)
Comment (piazzatron) [reply src/containers/explore-page/collections.ts:113]: yeah wasn't needed

File: src/containers/trending-underground/TrendingUndergroundPage.tsx (hunk 3 above)
Comment (raymondjacobson) [reply src/containers/trending-underground/TrendingUndergroundPage.tsx:100]: colocating is totally fine w/ me esp if you're the one to remove it. i think that if someone else becomes the one to gut the flags later, having them colocated is easier to trace

**Pull request merged**